### DISCUSSION TITLE
How to deep link to a Tab I've generated via addon

### DISCUSSION THREAD

**Discussion by steffchep** [storybookjs/storybook#15931] [Help]
So, I've created a custom addon to display an additional tab next to Canvas and Docs which will hold additional information for Designers. 
This works nicely including updating my URL to 
http://localhost:3333/?path=/design/components-example

However, when I copy and paste this link into a new tab, I get forwarded to the canvas tab of that story (address bar link gets changed accordingly to http://localhost:3333/?path=/story/components-example)

What do I have to do to make my new tab show up first if it is in the path?

Also, when Canvas or Docs is selected and I click on another story, the new story opens to that same tab. This doesn't happen when I have my custom tab open - new stories will open to the Canvas tab

I followed the instructions for [UI-based-addons](https://storybook.js.org/docs/react/addons/addon-types#ui-based-addons) to create my addon.

*edited for a little more clarity, I hope*

**Comment by steffchep** [comment]
Nobody has any ideas? I know it's somehow tied to routing and the viewModes, but I haven't figured out how to configure it.

**Comment by NsdHSO** [comment]
What exactly do you expect? More information please.

**Reply by steffchep** [reply]
I thought I had described it in my original post. Please let me know what specifically you need to know :)

In short, when I am looking at the new tab that I created with my addon, and change to another story, I want that story to ALSO show that new tab, as it does when I look at Docs or Canvas. Instead, the new story will open to the canvas.

The path in the Address bar does get set to my new viewMode when I click on my tab, as described above

**Comment by steffchep** [comment]
![storybook-design-tab](https://user-images.githubusercontent.com/10571670/131792967-c8b365c3-638e-4a2c-a801-12d9cd2fc08b.jpg)
![storybook-story-tab](https://user-images.githubusercontent.com/10571670/131792969-92242606-6e05-4990-b573-c0a9a02209f6.jpg)

**Comment by pazguille** [comment]
Same issue here!  When you have a custom TAB addon If copy/paste or reload the page, the current tabs and url back to /story

**Reply by steffchep** [reply]
From the discord discussion, it seems this is hard coded into storybook. Bummer, this is almost a show-stopper
